reasoner/Reasoner.java

```diff
@@ -95,7 +95,6 @@ private boolean mayReason(Disjunction disjunction) {
         for (Conjunction conj : disjunction.conjunctions()) {
             Set<Variable> vars = conj.variables();
             Set<Negation> negs = conj.negations();
-            if (iterate(vars).anyMatch(v -> v.resolvedTypes().isEmpty())) return true;
```

Comment:
type resolver no longer produces empty type hints, unless the query is inCoherent or a schema query